play-services-ads/core/src/main/java/com/google/android/gms/dynamite/descriptors/com/google/android/gms/ads/dynamite/ModuleDescriptor.java

```diff
@@ -16,9 +15,17 @@ public class ModuleDescriptor {
     public static final String MODULE_ID = "com.google.android.gms.ads.dynamite";
     public static final int MODULE_VERSION = 230500001;
 
-    public static void init(ContextWrapper context) {
-        Context baseContext = context.getBaseContext();
-        WebSettings.getDefaultUserAgent(baseContext);
-        Log.d("ModuleDescriptor", "init: context: " + baseContext);
+    /**
+     * The ads module might try to access the user agent, requiring initialization on the main thread,
+     * which may result in deadlocks when invoked from any other thread. This only happens with microG,
+     * because we don't use the highly privileged SELinux Sandbox that regular Play Services uses
+     * (which allows apps to read the user-agent from Play Services instead of the WebView). To prevent
+     * the issue we pre-emptively initialize the WebView.
+     */
+    public static void init(Context context) {
+        if (context instanceof ContextWrapper) {
+            context = ((ContextWrapper) context).getBaseContext();
+        }
+        WebSettings.getDefaultUserAgent(context);
```

Comment:
@mar-v-in
If I'm not wrong the User-Agent is changed by the profile.
Don't we need to run `ProfileManager.ensureInitialized`?

Reply:
This is running inside a dynamite module in the calling app's process. We can't change how the code inside the app accesses the user agent and we also don't have access to the profile manager from here. It also doesn't affect the user agent itself, it just makes sure the WebView is initialized when the client library code inside the app tries to access it later.